database/oracle/oracle.go

```diff
@@ -0,0 +1,446 @@
+package oracle
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"io/ioutil"
+	nurl "net/url"
+	"strings"
+
+	"github.com/godror/godror"
+
+	_ "github.com/godror/godror"
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+func init() {
+	db := Oracle{}
+	database.Register("oracle", &db)
+}
+
+const (
+	defaultMigrationsTable         = "SCHEMA_MIGRATIONS"
+	defaultStatementSeparator      = ";"
+	plsqlDefaultStatementSeparator = "---"
+	plsqlStatementEndToken         = "END;"
+)
+
+var (
+	ErrNilConfig      = fmt.Errorf("no config")
+	ErrNoDatabaseName = fmt.Errorf("no database name")
+)
+
+type Config struct {
+	MigrationsTable         string
+	DisableMultiStatements  bool
```

Comment:
> In my use case, I hope I can write the migration file like Postgres do, i.e, write multi-statements in a single file, so I prefer to enable multi-statements support by default, and meanwhile, the user can disable multi-statements via x-disable-multi-statements=true

Do normal statements differ from PL/SQL statements? If so, we should support multi statements for each separately.

You're welcome to have your preference, but don't impose that on others and instead follow the defaults and patterns already set by the project.

Multiple statements in a migration work out of the box for postgres b/c the pq driver works that way. Likewise, for mysql since we're not parsing the migration. However, multiple statements don't work out of the box for cassandra since that's a reflection of the golang cassandra driver.
Namely, if the migration is not being fed directly into the underlying DB driver (e.g. is parsed by migrate), then it needs to be a configurable option that's off by default.

Reply:
resolved